packages/gatsby-source-contentful/src/fetch.js

```diff
@@ -34,6 +34,11 @@ module.exports = async ({ syncToken, reporter, pluginConfig }) => {
     locales = await client.getLocales().then(response => response.items)
     defaultLocale = _.find(locales, { default: true }).code
     locales = locales.filter(pluginConfig.get(`localeFilter`))
+    if (locales.length === 0) {
+      reporter.panic(
+        `No locales found to build data model. Plase check if your localeFilter configuration is configured properly.`
```

Comment:
Could we also print what `locales` we get from Contentful space (from the `client.getLocales()` response)? So user/developer have some context why their `localeFilter` might be wrong (and don't have to add console.log to their filter) - this would just serve as mean to reduce feedback loop.

Implementation wise this would pretty much mean storing `client.getLocales().[...]` response to different variable so we don't overwrite it with the `localeFilter` and can display it

Reply:
Little bit more verbose than I expected, but it's been added.